engine/src/in_joystick.cpp

```diff
@@ -175,58 +179,69 @@ void InitJoystick() {
             }
         }
     }
+    num_joysticks = 0;
+
 #ifndef NO_SDL_JOYSTICK
 #ifdef HAVE_SDL
-    num_joysticks = SDL_NumJoysticks();
-    VS_LOG(info, (boost::format("%1% joysticks were found.\n\n") % num_joysticks));
-    VS_LOG(info, "The names of the joysticks are:\n");
+    SDL_JoystickID *joysticks = SDL_GetJoysticks(&num_joysticks);
+    if (num_joysticks > MAX_JOYSTICKS) {
+        num_joysticks = MAX_JOYSTICKS;
+    }
+    VS_LOG(important_info, (boost::format("%1% joysticks were found.\n\n") % num_joysticks));
+    VS_LOG(important_info, "The names of the joysticks are:\n");
+    if (joysticks) {
+        for (int i1 = 0; i1 < MAX_JOYSTICKS; ++i1) {
+            if (i1 < num_joysticks) {
+                const SDL_JoystickID instance_id = joysticks[i1];
+                VS_LOG(important_info, (boost::format("    %1%\n") % SDL_GetJoystickNameForID(instance_id)));
+                joystick[i1] = new JoyStick(i1, instance_id);
+            } else {
+                joystick[i1] = new JoyStick(i1, 0);
+            }
+        }
+    }
+    SDL_free(joysticks);
+
 #else
```

Comment:
Remove this. We no longer have an option of running the game without SDL. We pick a library and stick to it.